doc/arkode/guide/source/Usage/SPRKStep_c_interface/User_callable.rst

```diff
@@ -0,0 +1,1229 @@
+   :param hcur: step size to be attempted on the next internal step.
+
+   :return:
+      * *ARK_SUCCESS* if successful
+      * *ARK_MEM_NULL* if the SPRKStep memory was ``NULL``
+
+
+.. c:function:: int SPRKStepGetCurrentTime(void* arkode_mem, realtype* tcur)
+
+   Returns the current internal time reached by the solver.
+
+   :param arkode_mem: pointer to the SPRKStep memory block.
+   :param tcur: current internal time reached.
+
+   :return:
+      * *ARK_SUCCESS* if successful
+      * *ARK_MEM_NULL* if the SPRKStep memory was ``NULL``
+
+
+.. c:function:: int SPRKStepGetCurrentState(void *arkode_mem, N_Vector *ycur)
+
+   Returns the current internal solution reached by the solver.
+
+   **Arguments:**
+      * *arkode_mem* -- pointer to the ARKStep memory block.
+      * *ycur* -- current internal solution.
+
+   **Return value:**
+      * *ARK_SUCCESS* if successful
+      * *ARK_MEM_NULL* if the ARKStep memory was ``NULL``
+
+   **Notes:**
+      Users should exercise extreme caution when using this function,
+      as altering values of *ycur* may lead to undesirable behavior, depending
+      on the particular use case and on when this routine is called.
+
+
+.. c:function:: int SPRKStepGetStepStats(void* arkode_mem, long int* nsteps, realtype* hinused, realtype* hlast, realtype* hcur, realtype* tcur)
+
+   Returns many of the most useful optional outputs in a single call.
+
+   :param arkode_mem: pointer to the SPRKStep memory block.
+   :param nsteps: number of steps taken in the solver.
+   :param hinused: actual value of initial step size.
+   :param hlast: step size taken on the last internal step.
+   :param hcur: step size to be attempted on the next internal step.
+   :param tcur: current internal time reached.
+
+   :return:
+      * *ARK_SUCCESS* if successful
+      * *ARK_MEM_NULL* if the SPRKStep memory was ``NULL``
+
+
+.. c:function:: int SPRKStepPrintAllStats(void* arkode_mem, FILE* outfile, SUNOutputFormat fmt)
+
+   Outputs all of the integrator statistics.
+
+     * *arkode_mem* -- pointer to the SPRKStep memory block.
+     * *outfile* -- pointer to output file.
+     * *fmt* -- the output format:
+
+       * :c:enumerator:`SUN_OUTPUTFORMAT_TABLE` -- prints a table of values
+       * :c:enumerator:`SUN_OUTPUTFORMAT_CSV` -- prints a comma-separated list
+         of key and value pairs e.g., ``key1,value1,key2,value2,...``
+
+   :return:
+     * *ARK_SUCCESS* -- if the output was successfully.
+     * *CV_MEM_NULL* -- if the SPRKStep memory was ``NULL``.
+     * *CV_ILL_INPUT* -- if an invalid formatting option was provided.
+
+   .. note::
+
+      The file ``scripts/sundials_csv.py`` provides python utility functions to
+      read and output the data from a SUNDIALS CSV output file using the key
+      and value pair format.
+
+
+
+.. c:function:: char *SPRKStepGetReturnFlagName(long int flag)
+
+   Returns the name of the SPRKStep constant corresponding to *flag*.
```

Comment:
Adding reference to the table of constants would be helpful here

Reply:
Indeed, Ill ad this in the other steppers too.